zk-token-sdk/src/instruction/batched_grouped_ciphertext_validity.rs

```diff
@@ -0,0 +1,206 @@
+        let pod_destination_pubkey = pod::ElGamalPubkey(destination_pubkey.to_bytes());
+        let pod_auditor_pubkey = pod::ElGamalPubkey(auditor_pubkey.to_bytes());
+        let pod_grouped_ciphertext_lo = (*grouped_ciphertext_lo).into();
+        let pod_grouped_ciphertext_hi = (*grouped_ciphertext_hi).into();
+
+        let context = BatchedGroupedCiphertext2HandlesValidityProofContext {
+            destination_pubkey: pod_destination_pubkey,
+            auditor_pubkey: pod_auditor_pubkey,
+            grouped_ciphertext_lo: pod_grouped_ciphertext_lo,
+            grouped_ciphertext_hi: pod_grouped_ciphertext_hi,
+        };
+
+        let mut transcript = context.new_transcript();
+
+        let proof = AggregatedValidityProof::new(
+            (destination_pubkey, auditor_pubkey),
+            (amount_lo, amount_hi),
+            (opening_lo, opening_hi),
+            &mut transcript,
+        )
+        .into();
+
+        Ok(Self { context, proof })
+    }
+}
+
+impl ZkProofData<BatchedGroupedCiphertext2HandlesValidityProofContext>
+    for BatchedGroupedCiphertext2HandlesValidityProofData
+{
+    const PROOF_TYPE: ProofType = ProofType::BatchedGroupedCiphertext2HandlesValidity;
+
+    fn context_data(&self) -> &BatchedGroupedCiphertext2HandlesValidityProofContext {
+        &self.context
+    }
+
+    #[cfg(not(target_os = "solana"))]
+    fn verify_proof(&self) -> Result<(), ProofError> {
+        let mut transcript = self.context.new_transcript();
+
+        let destination_pubkey = self.context.destination_pubkey.try_into()?;
+        let auditor_pubkey = self.context.auditor_pubkey.try_into()?;
+        let grouped_ciphertext_lo: GroupedElGamalCiphertext<2> =
+            self.context.grouped_ciphertext_lo.try_into()?;
+        let grouped_ciphertext_hi: GroupedElGamalCiphertext<2> =
+            self.context.grouped_ciphertext_hi.try_into()?;
+
+        let destination_handle_lo = grouped_ciphertext_lo.handles.get(0).unwrap();
+        let auditor_handle_lo = grouped_ciphertext_lo.handles.get(1).unwrap();
+
+        let destination_handle_hi = grouped_ciphertext_hi.handles.get(0).unwrap();
+        let auditor_handle_hi = grouped_ciphertext_hi.handles.get(1).unwrap();
+
+        let proof: AggregatedValidityProof = self.proof.try_into()?;
+
+        proof
+            .verify(
+                (&destination_pubkey, &auditor_pubkey),
+                (
+                    &grouped_ciphertext_lo.commitment,
+                    &grouped_ciphertext_hi.commitment,
+                ),
+                (destination_handle_lo, destination_handle_hi),
+                (auditor_handle_lo, auditor_handle_hi),
+                &mut transcript,
+            )
+            .map_err(|e| e.into())
+    }
+}
+
+#[cfg(not(target_os = "solana"))]
+impl BatchedGroupedCiphertext2HandlesValidityProofContext {
+    fn new_transcript(&self) -> Transcript {
+        let mut transcript = Transcript::new(b"AggregateCiphertextValidityProof");
```

Comment:
```suggestion
        let mut transcript = Transcript::new(b"BatchedGroupedCiphertextValidityProof");
```
I'm not sure how much this matters as long as it's unique, but might as well be consistent?

Reply:
Ah sorry I missed it. Thanks for the catch!